File: src/plugin/interaction/combat/magic/ancient/smoke/SmokeBlitzSpellPlugin.java
package plugin.interaction.combat.magic.ancient.smoke;

import org.redrune.game.content.combat.player.style.MagicCombatStyle;
import org.redrune.game.entity.actor.Actor;
import org.redrune.game.entity.actor.player.Player;
import org.redrune.game.content.plugin.combat.spell.type.CombatSpellPlugin;
import org.redrune.game.entity.projectile.ProjectileManager;
import org.redrune.utility.functions.RandomFunction;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 8/1/2017
 */
public class SmokeBlitzSpellPlugin implements CombatSpellPlugin {
	
	@Override
	public int delay(Player player) {
		return 4;
	}
	
	@Override
	public int animationId() {
		return 1978;
	}
	
	@Override
	public int hitGfx() {
		return 387;
	}
	
	@Override
	public int maxHit(Player player, Actor target) {
		return 230;
	}
	
	@Override
	public void cast(Player source, Actor target, MagicCombatStyle style) {
		ProjectileManager.sendProjectile(ProjectileManager.createSpeedDefinedProjectile(source, target, 386, 18, 9, 52, 15, 0));
		style.sendSpell(source, target, this, null, null).consume(spellContext -> {
			if (spellContext.getHit().getDamage() != 0 && RandomFunction.percentageChance(10)) {
				Actor spellTarget = spellContext.getTarget();
				if (!spellTarget.getPoisonManager().isPoisoned()) {
					spellTarget.getPoisonManager().makePoisoned(40);
				}
			}
		});
	}
	
	@Override
	public int spellId() {
		return 29;
	}
	
	@Override
	public double exp() {
		return 42;
	}
	
	@Override
	public MagicBook book() {
		return MagicBook.ANCIENTS;
	}
	
}

File: src/main/java/plugin/interaction/combat/magic/ancient/smoke/SmokeBlitzSpellPlugin.kt
package plugin.interaction.combat.magic.ancient.smoke

import game.content.plugin.combat.spell.type.CombatSpellPlugin
import game.entity.actor.Actor
import game.content.entity.actor.combat.player.style.MagicCombatStyle
import game.entity.projectile.ProjectileManager
import utility.constants.MagicConstants.MagicBook
import game.content.entity.actor.combat.CombatSwingDetail
import utility.functions.RandomFunction
import game.entity.actor.player.Player

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 8/1/2017
 */
class SmokeBlitzSpellPlugin : CombatSpellPlugin {
    override fun delay(player: Player): Int {
        return 4
    }

    override fun animationId(): Int {
        return 1978
    }

    override fun hitGfx(): Int {
        return 387
    }

    override fun maxHit(player: Player, target: Actor): Int {
        return 230
    }

    override fun cast(source: Player, target: Actor, style: MagicCombatStyle) {
        ProjectileManager.sendProjectile(
            ProjectileManager.createSpeedDefinedProjectile(
                source,
                target,
                386,
                18,
                9,
                52,
                15,
                0
            )
        )
        style.sendSpell(source, target, this, null, null).consume { spellContext: CombatSwingDetail ->
            if (spellContext.hit.damage != 0 && RandomFunction.percentageChance(10)) {
                val spellTarget = spellContext.target
                if (!spellTarget.poisonManager.isPoisoned) {
                    spellTarget.poisonManager.makePoisoned(40)
                }
            }
        }
    }

    override fun spellId(): Int {
        return 29
    }

    override fun exp(): Double {
        return 42.0
    }

    override fun book(): MagicBook {
        return MagicBook.ANCIENTS
    }

    override fun castSoundId(): Int {
        return 183
    }

    override fun impactSoundId(): Int {
        return 184
    }
}